components/src/core/app-bar/app-bar.component.ts

```diff
@@ -0,0 +1,169 @@
+import {
+    AfterViewInit,
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    ElementRef,
+    Input,
+    OnChanges,
+    OnInit,
+    SimpleChanges,
+    ViewChild,
+    ViewEncapsulation,
+} from '@angular/core';
+import { fromEvent, interval } from 'rxjs';
+import { throttle } from 'rxjs/operators';
+
+@Component({
+    selector: 'pxb-app-bar',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    encapsulation: ViewEncapsulation.None,
+    styleUrls: ['./app-bar.component.scss'],
+    template: `
+        <mat-toolbar
+            #pxbAppBar
+            color="primary"
+            class="pxb-app-bar-content"
+            [style.height.px]="currentHeight"
+            [style.marginTop.px]="getMarginTop()"
+            [class.pxb-app-bar-sticky]="collapsedHeight === currentHeight || mode !== 'dynamic'"
+        >
+            <mat-toolbar-row>{{ currentHeight }}</mat-toolbar-row>
```

Comment:
Can we generalize this to accept any content? I'm curious if you'll run into the same jittery behavior as I am when there's more than just a string in here.

Reply:
Yeah I'll added a temporary ng-content so you can add anything into the toolbar content. 

Whenever the 3-liner component is added, this will probably be a templateRef instead of a ng-content.